source/boundary_velocity/ascii_data.cc

```diff
@@ -107,7 +121,11 @@ namespace aspect
           {
             this->scale_factor               /= year_in_seconds;
           }
-
+        prm.enter_subsection("Ascii data model");
+        {
+          use_spherical_unit_vectors = prm.get_bool("Use spherical unit vectors");
```

Comment:
Maybe you can add an AssertThrow that when this is set to true, you check whether the geometry model is not a box?